modules/stochastic_tools/src/surrogates/CovarianceFunctionBase.C

```diff
@@ -16,10 +16,72 @@ CovarianceFunctionBase::validParams()
   params.addClassDescription("Base class for covariance functions");
   params.registerBase("CovarianceFunctionBase");
   params.registerSystemAttributeName("CovarianceFunctionBase");
+  params.addParam<std::vector<Real>>("length_factor", "Length Factor to use for Covariance Kernel");
+  params.addParam<Real>("signal_variance",
+                        "Signal Variance ($\\sigma_f^2$) to use for kernel calculation.");
+  params.addParam<Real>("noise_variance",
+                        "Noise Variance ($\\sigma_n^2$) to use for kernel calculation.");
   return params;
 }
 
 CovarianceFunctionBase::CovarianceFunctionBase(const InputParameters & parameters)
-  : MooseObject(parameters)
+  : MooseObject(parameters),
+    _length_factor(getParam<std::vector<Real>>("length_factor")),
+    _sigma_f_squared(getParam<Real>("signal_variance")),
+    _sigma_n_squared(getParam<Real>("noise_variance"))
 {
 }
+
+void
+CovarianceFunctionBase::buildHyperParamMap(
+    std::unordered_map<std::string, Real> & map,
+    std::unordered_map<std::string, std::vector<Real>> & vec_map) const
+{
+  map["noise_variance"] = _sigma_n_squared;
+  map["signal_variance"] = _sigma_f_squared;
+
+  vec_map["length_factor"] = _length_factor;
+}
+
+void
+CovarianceFunctionBase::loadHyperParamMap(
+    std::unordered_map<std::string, Real> & map,
+    std::unordered_map<std::string, std::vector<Real>> & vec_map)
+{
+  _sigma_n_squared = map["noise_variance"];
+  _sigma_f_squared = map["signal_variance"];
+
+  _length_factor = vec_map["length_factor"];
+}
+
+bool
+CovarianceFunctionBase::isTunable(std::string name) const
+{
+  if (_tunable_hp.find(name) != _tunable_hp.end())
+    return true;
+  else if (isParamValid(name))
+    ::mooseError("Tuning not supported for parameter ", name);
+  else
+    ::mooseError("Parameter ", name, " selected for tuning is not a valid parameter");
```

Comment:
I included the error exist because if a user types into the tunabel list wrong sigmal_variance instead of signal_variance, they get en error directing them to look, rather than just not throwing anything. This little framework specifies the parameters you wish to tune as a list, so it bypasses some of the standard parameter error checking.